mods/variations/init.lua

```diff
@@ -0,0 +1,99 @@
+variations = {}
+
+variations.variations = {
+	{
+		name = "brick";
+		description = "Brick";
+		texture = "0,0";
+	},
+	{
+		name = "big_tile";
+		description = "Big Tile";
+		texture = "1,0";
+	},
+	{
+		name = "tile";
+		description = "Tile";
+		texture = "2,0";
+	},
+	{
+		name = "tall_tile";
+		description = "Tall Tile";
+		texture = "0,1";
+	},
+	{
+		name = "small_brick";
+		description = "Small Brick";
+		texture = "1,1";
+	},
+	{
+		name = "small_tile";
+		description = "Small Tile";
+		texture = "2,1";
+	},
+	{
+		name = "horizontal_brick";
+		description = "Horizontal Brick";
+		texture = "0,2";
+	},
+	{
+		name = "cross_tile";
+		description = "Cross Tile";
+		texture = "1,2";
+	},
+	{
+		name = "spiral_tile";
+		description = "Spiral Tile";
+		texture = "2,2";
+	},
+}
+
+function variations.register_for_base(base_node, transparent)
+	local base_definition = minetest.registered_nodes[base_node]
+	for _, variation in ipairs(variations.variations) do
+		-- We get an iterator function over substrings split by :
+		local base_node_full_name = string.gmatch(base_node, "([^:]+)")
+		-- First call will give us the modname
+		-- local base_node_modname = base_node_full_name()
+		-- Unused variable warning...
+		base_node_full_name()
+		-- and the second will give us the node name
+		local base_node_name = base_node_full_name()
```

Comment:
potential duplicates above?

Reply:
Returns modname, so the second one returns nodename

Reply:
unnecessary complication?
Use `local sname = string.match(name, ':(.*)')` to get the node name straight away?